### ISSUE TITLE
adding support for real time sensor data monitoring and  sensors based conditional automations

### ISSUE THREAD

**Issue by charithmadhuranga** [gardenifi/raspirri_app#7]
adding graphs to plot real-time sensor data sent through MQTT (eg - temperature, humidity) with support for conditional automation.

**Comment by mariosk** [comment]
Hi @charithmadhuranga, thanks for your request, but there is no such sensor on the RaspirriV1 board!

**Comment by charithmadhuranga** [comment]
Hi @mariosk  if there are no sensors how can the system determine the optimal time and amount of irrigation?If we use a weather api it will not represent the ideal conditions as it depends on the condition of the environment someone deploys above system.

**Comment by charithmadhuranga** [comment]
I assume the above RaspirriV1 board is not a custom pcb but a general raspberry pi computer. if that is the case then extending the backend with plug-in system will give anyone to add more features to the system  not only any kind of sensors but cameras as well.

**Comment by mariosk** [comment]
You're right @charithmadhuranga! RaspirriV1 board is not a custom PCB, it's a Raspberry PI 4 with a custom electronics circuit to support 24v electrovalves. You're also right if we need to support optimal time and irrigation amount we definitely need to add extra sensors for humidity, temperature, etc. The current implementation supports only scheduled programs by the user of the mobile app.
